Relative path include in .gitconfig throws AssertionError

My `.gitconfig` contains:

```
[include]
    path = .gitconfig_common
```

According to the [Git documentation](https://git-scm.com/docs/git-config#_includes), this is allowed to be a _relative path_:

> If the value of the variable is a relative path, the path is considered to be relative to the configuration file in which the include directive was found.

However, GitPython throws:

```
  File "/Users/daniel/.cache/uv/archive-v0/YvUvYFO1C0vkI8VZmPICW/lib/python3.12/site-packages/briefcase/commands/base.py", line 1365, in get_git_config_value
    if git_config.has_option(section, option):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daniel/.cache/uv/archive-v0/YvUvYFO1C0vkI8VZmPICW/lib/python3.12/site-packages/git/config.py", line 113, in assure_data_present
    self.read()
  File "/Users/daniel/.cache/uv/archive-v0/YvUvYFO1C0vkI8VZmPICW/lib/python3.12/site-packages/git/config.py", line 654, in read
    assert osp.isabs(file_path), "Need absolute paths to be sure our cycle checks will work"
           ^^^^^^^^^^^^^^^^^^^^
AssertionError: Need absolute paths to be sure our cycle checks will work
```

The workaround is to modify my `.gitconfig` and make the path absolute, but I'd like it to work without modifying my `.gitconfig` to suit GitPython.

--- a/git/config.py
+++ b/git/config.py
@@ -634,6 +634,8 @@ class GitConfigParser(cp.RawConfigParser, metaclass=MetaParserBuilder):
             files_to_read = list(self._file_or_files)
         # END ensure we have a copy of the paths to handle
 
+        files_to_read = [osp.abspath(path) if isinstance(path, (str, os.PathLike)) else path for path in files_to_read]
+
         seen = set(files_to_read)
         num_read_include_files = 0
         while files_to_read:

--- a/test/test_config.py
+++ b/test/test_config.py
@@ -311,6 +311,20 @@ class TestBase(TestCase):
             check_test_value(cr, tv)
 
     @with_rw_directory
+    def test_config_relative_path_include(self, rw_dir):
+        included_path = osp.join(rw_dir, "included")
+        with GitConfigParser(included_path, read_only=False) as cw:
+            cw.set_value("included", "value", "included")
+
+        config_path = osp.join(rw_dir, "config")
+        with GitConfigParser(config_path, read_only=False) as cw:
+            cw.set_value("include", "path", "included")
+
+        relative_config_path = osp.relpath(config_path)
+        with GitConfigParser(relative_config_path, read_only=True) as cr:
+            assert cr.get_value("included", "value") == "included"
+
+    @with_rw_directory
     def test_multiple_include_paths_with_same_key(self, rw_dir):
         """Test that multiple 'path' entries under [include] are all respected.
 

